pkg/cmd/pulumi/state_edit_test.go

```diff
@@ -86,3 +86,48 @@ func TestSnapshotFrontendRoundTrip(t *testing.T) {
 	// The round-tripped snapshot text should be the same as the original.
 	assert.Equal(t, text, roundTrippedText)
 }
+
+func TestOpenInEditorMultiPart(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name      string
+		command   string
+		filename  string
+		wantError bool
+	}{
+		{
+			name:     "Simple command",
+			command:  "echo",
+			filename: "filename.txt",
+		},
+		{
+			name:     "Command with arguments",
+			command:  "echo Hello",
+			filename: "filename.txt",
+		},
+		{
+			name:     "Complex command",
+			command:  "echo --foo --bar a",
+			filename: "filename.txt",
+		},
+		{
+			name:      "Invalid command",
+			command:   "",
+			filename:  "should-not-exist-file.txt",
+			wantError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := openInEditorInternal(tt.command, tt.filename)
+			if tt.wantError {
+				assert.Error(t, err)
```

Comment:
If checking for errors you really ought to use ErrorContains, it's far to easy to write a test that errors but isn't actually checking the code path for the error you want.

Reply:
kk. changing the `""` check to panic means that they should not error and should simplify this.